cmd/node/flags.go

```diff
@@ -507,26 +517,66 @@ func getWorkingDir(workingDir string, log logger.Logger) string {
 }
 
 func applyCompatibleConfigs(log logger.Logger, configs *config.Configs) error {
-	importDbFlags := configs.ImportDbConfig
-	importDbFlags.ImportDbNoSigCheckFlag = importDbFlags.ImportDbNoSigCheckFlag && importDbFlags.IsImportDBMode
-	importDbFlags.ImportDbSaveTrieEpochRootHash = importDbFlags.ImportDbSaveTrieEpochRootHash && importDbFlags.IsImportDBMode
+	if configs.FlagsConfig.EnablePprof {
+		runtime.SetMutexProfileFraction(5)
+	}
 
-	if importDbFlags.IsImportDBMode {
-		return processConfigImportDBMode(log, configs)
+	operationModes := strings.Split(configs.FlagsConfig.OperationMode, ",")
+	err := checkOperationModes(operationModes)
+	if err != nil {
+		return err
+	}
+
+	isInImportDBMode := configs.ImportDbConfig.IsImportDBMode || sliceContainsElement(operationModes, operationModeImportDb)
+	if isInImportDBMode {
+		err := processConfigImportDBMode(log, configs)
+		if err != nil {
+			return err
+		}
 	}
 
 	// if FullArchive is enabled, we override the conflicting StoragePruning settings and StartInEpoch as well
-	if configs.PreferencesConfig.Preferences.FullArchive {
-		return processConfigFullArchiveMode(log, configs)
+	isInFullArchiveMode := configs.PreferencesConfig.Preferences.FullArchive || sliceContainsElement(operationModes, operationModeFullArchive)
+	if isInFullArchiveMode {
+		processConfigFullArchiveMode(log, configs)
 	}
 
-	if configs.FlagsConfig.EnablePprof {
-		runtime.SetMutexProfileFraction(5)
+	isInHistoricalBalancesMode := sliceContainsElement(operationModes, operationModeHistoricalBalances)
+	if isInHistoricalBalancesMode {
+		processHistoricalBalancesMode(log, configs)
+	}
+
+	isInDbLookupExtensionMode := sliceContainsElement(operationModes, operationModeDbLookupExtension)
+	if isInDbLookupExtensionMode {
+		processDbLookupExtensionMode(log, configs)
 	}
 
 	return nil
 }
 
+func processHistoricalBalancesMode(log logger.Logger, configs *config.Configs) {
+	configs.GeneralConfig.StoragePruning.ObserverCleanOldEpochsData = false
```

Comment:
For validators, as well, perhaps? (in case there are validators started with this operation mode - only if we can recommend this, otherwise not).

Reply:
done